File: 2021/day-1/part-2/main.py
def main(input: str) -> int:
    data = list(map(int, open(input).readlines())); count = 0
    for index, value in enumerate(data):
        if index == len(data)-3: break
        if (value + data[index+1] + data[index+2]) < (data[index+1] + data[index+2] + data[index+3]):
                count +=1
    return count

if __name__ == "__main__":
    result = main("./inputs.txt"); print(result)

File: 2021/day-1/part-2/main.jl
function main()
    data = readlines(open("inputs.txt", "r"));
    data = parse.(Int, data);
    count = 0;
    for (index, line) in enumerate(data)
        if index == length(data)-2 
            break 
        end
        if line + data[index+1] + data[index+2] < data[index+1] + data[index+2] + data[index+3]
            count += 1;
        end
    end
    return count;
end
println(main())